### PR TITLE
[AIRFLOW-4785] Don't apply limits on running DummyOperator

### PR DESCRIPTION
**Pull request by tooptoop4**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -386,8 +386,8 @@ def __init__(
         self.retries = retries
         self.queue = queue
         self.pool = pool
-        self.pool_slots = pool_slots
-        if self.pool_slots < 1:
+        self.pool_slots = 0 if self.__class__.__name__ == 'DummyOperator' else pool_slots
+        if self.pool_slots < 0 or (self.pool_slots == 0 and self.__class__.__name__ != 'DummyOperator'):
```
Comment (mik-laj) [inline airflow/models/baseoperator.py:390]: Will this work with ExternalTaskMarker?

File: airflow/ti_deps/deps/pool_slots_available_dep.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def _get_dep_statuses(self, ti, session, dep_context=None):
         if ti.state in STATES_TO_COUNT_AS_RUNNING:
             open_slots += ti.pool_slots
 
-        if open_slots <= (ti.pool_slots - 1):
+        if open_slots < ti.pool_slots:
```
Comment (mik-laj) [inline airflow/ti_deps/deps/pool_slots_available_dep.py:66]: Could you say a little more about this change?

File: airflow/models/baseoperator.py (hunk 1 above)
Comment (tooptoop4) [reply airflow/models/baseoperator.py:390]: how is ExternalTaskMarker relevant? admittedly i've never heard of it!

File: airflow/ti_deps/deps/pool_slots_available_dep.py (hunk 2 above)
Comment (tooptoop4) [reply airflow/ti_deps/deps/pool_slots_available_dep.py:66]: My understanding is: open_slots means how many slots left in the pool (ie available to be consumed/tasks run in), pool_slots means how many slots this specific taskinstance aims to consume. Previously it assumed each task would take at least 1 slot, now I'm making it handle case where task can take 0 slots

File: airflow/models/baseoperator.py (hunk 1 above)
Comment (mik-laj) [reply airflow/models/baseoperator.py:390]:
```
class ExternalTaskMarker(DummyOperator):
```

Comment (tooptoop4) [reply airflow/models/baseoperator.py:390]: what do u suggest? I think its fine for ExternalTaskMarker not to consume a slot

Comment (mik-laj) [reply airflow/models/baseoperator.py:390]: Does this change make sense at all if my PR has been merged?  It seems to me that the condition will never be met.
https://github.com/apache/airflow/pull/7880

**Pull request closed without merging**